types/basetypes/float64_value.go

```diff
@@ -68,7 +70,7 @@ func NewFloat64Unknown() Float64Value {
 func NewFloat64Value(value float64) Float64Value {
 	return Float64Value{
 		state: attr.ValueStateKnown,
-		value: value,
+		value: big.NewFloat(value),
```

Comment:
This will panic if [`math.Nan()`](https://pkg.go.dev/math#NaN) is provided here. This problem currently exists with the underlying value as `float64`, the panic will just come later when `tftypes.ValidateValue` is called.

Are there any non-breaking changes we can make to smooth this out?

Reply:
Sorry I'm being dense, are you saying this is a new problem or existing problem? I'm not sure we could do anything different here, other than potentially raise our own panic message.

Reply:
This is an existing problem today, if today you write:

```go
data.FloatAttribute = types.NewFloat64Value(math.Nan())
```

You will get a panic later when `tftypes` attempts to use that float64, I think the first place it happens is with [`tftypes.ValidateValue`](https://github.com/hashicorp/terraform-plugin-framework/blob/f937d7456b05f45e8e8d7fb8dcb6bda806fd9861/types/basetypes/float64_value.go#L118).

Panic raised here:
https://github.com/hashicorp/terraform-plugin-go/blob/81f0d496aaa39e43cc185f5c961df5fcd86192da/tftypes/primitive.go#L347 


When we change this logic, a developer would receive a panic immediately:
```go
// Panic right away! :O
data.FloatAttribute = types.NewFloat64Value(math.Nan())
```

Reply:
So the same result (a panic error), but at different points. I can create a separate issue for this if we don't want to handle it now

Reply:
Sounds good, added doc note in 8672e73 ⚡